hybris/egl/platforms/common/support.h

```diff
@@ -11,7 +11,7 @@
  *
  */
 #define container_of(ptr, type, member) ({			\
-	const typeof( ((type *)0)->member ) *__mptr = (ptr);	\
+	const decltype( ((type *)0)->member ) *__mptr = (ptr);	\
```

Comment:
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I/home/android_head/ -I../../../include -I../../../egl -I/home/android_head/ -I../../../common/ -I/usr/local/include -I/usr/local/include -I/home/and
roid_head/ -ggdb -O0 -g -O2 **-std=c++14** -MT libhybris_eglplatformcommon_la-server_wlegl_buffer.lo -MD -MP -MF .deps/libhybris_eglplatformcommon_la-server_wlegl_buffer.Tpo -c server_wlegl_buffer.cpp  -fPIC 
-DPIC -o .libs/libhybris_eglplatformcommon_la-server_wlegl_buffer.o
In file included from nativewindowbase.cpp:13:0:
nativewindowbase.cpp: In static member function 'static void BaseNativeWindowBuffer::_decRef(android_native_base_t*)':
support.h:14:14: error: ISO C++ forbids declaration of 'typeof' with no type [-fpermissive]
  const typeof( ((type *)0)->member ) *__mptr = (ptr); \


Because I met this error, did not you ?  It is strange that my ISO C++ can't recognize typeof but yours can't recognize decltype

Reply:
After I searched some article, I think we should used the decltype instead of typeof.

Could you please show your gcc/g++ version ?
here is mine
gcc version 7.4.0 (Ubuntu 7.4.0-1ubuntu1~18.04.1)

Reply:
Because typeof is a gcc extension. You need to use -std=gnu++11. The GCC in Sailfish is currently is GCC 4.9

Reply:
but in configure.ac I add this:
AX_CHECK_COMPILE_FLAG([-std=c++14], [CXXFLAGS="$CXXFLAGS -std=c++14"])

So every thing should be built with -std=c++14

And why the "decltype" doesn't work in your environment even with C++14?

Reply:
GCC 4.9 doesn't support C++14 so you can't use that here.